### DISCUSSION TITLE
Custom field labels

### DISCUSSION THREAD

**Discussion by julbd** [directus/directus#13174] [Feature Requests]
For now, the only way to change a field name is to re-create it or to create language-specific translations.

This label would be a simpler way to customize and update the displayed field name independently from database structure or user language, closer to others properties behaviors. It implies :
* A new `label` property for fields. When empty (default), it defaults to the formatted field name like now.
* Field name translations become field label translations and works like `Notes` (a global value + optional translations).

As a first contribution, I can try to implement it if you are okay with it 😄 

UX Suggestion :
<img width="857" alt="Screenshot 2022-05-08 at 14 19 40" src="https://user-images.githubusercontent.com/64699408/167295899-fe8c6bf2-a318-4fb5-b8ef-d832d5e3e2ae.png">.

**Comment by azrikahar** [comment]
Coincidentally there's actually a pretty similar discussion brought up over at #12868! Feel free to take the points over there into consideration as well, or perhaps add on any thoughts you may have based on them 👍

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by Geoffrey60fps** [reply]
Personally, i can only upvote when I encounter a bug/problem and find the relevant issue, which is the case today.
I don't think alot of user get a mail when there is a new issue on github when they're just user of the application.

Having an easy way to set a label for a collection/field/anything would really be a plus in terms of directus adoption.

**Reply by rijkvanzanten** [reply]
> Personally, i can only upvote when I encounter a bug/problem and find the relevant issue, which is the case today.
I don't think alot of user get a mail when there is a new issue on github when they're just user of the application.

I recognize that, but leaving everything open indefinitely doesn't work at the scale that we're operating at on this repo. What happened before was is that people generally wouldn't find the already open feature request that applied to their need, and either ignored it or created a duplicate. At one point, we had more open feature requests (of which mostly duplicates and not-filled out posts) than GitHub was able to display (it only shows the first 1000 open posts, and hides the rest 🙈) which made that problem way worse. This new approach has really turned things around in feature requests and has allowed us to properly triage and prioritize based on community interest, whereas before it grew to a scale where it basically became impossible to seriously engage with community feature requests 🙂

**Comment by julbd** [comment]
@rijkvanzanten hmm… I understand the need to prioritize, but I'm not sure this is a relevant way to choose features to implement. Shiny and complex features will receive much more attention than the ones that bring small experience improvements… but the latter are still necessary to make a good product.

**Comment by kevlarr** [comment]
Agree with @julbd. It isn't big or shiny but having human-friendly labels that don't directly map to the database column name has been a fairly standard practice for decades and I'm surprised to see it not supported and closed so quickly. Especially since other requests took several years to get upvotes, e.g. https://github.com/directus/directus/discussions/2711 - 90 days is truly not long.

**Reply by rijkvanzanten** [reply]
> and I'm surprised to see it not supported 

It is; you can set a custom label that's used in the app for each of the app's supported languages 👍🏻 


> Especially since other requests took several years to get upvotes

That's not entirely accurate. Those requests already had tens/hundreds of upvotes when we switched to this updated process, so we didn't close them out. 

> 90 days is truly not long

Agree to disagree. There have been a lot of feature requests that have easily hit that initial upvotes requirement. This system has been way better at making sure we review and prioritize feature requests that are appropriate for a bigger part of the community than having everything be an open void to scream in like it was before 👍🏻

**Reply by kevlarr** [reply]
> It is; you can set a custom label that's used in the app for each of the app's supported languages

Ah, I had mistakenly remembered it as showing the language in the label, but now I'm realizing that it was only showing the language in the modal where setting up the translation. That's fine.

> Agree to disagree. There have been a lot of feature requests that have easily hit that initial upvotes requirement. This system has been way better at making sure we review and prioritize feature requests that are appropriate for a bigger part of the community than having everything be an open void to scream in like it was before 👍🏻

I guess. Not my ship to sail, so.. whatever works for you all. 🤷

**Comment by yonnic** [comment]
its the end of 2023 and its still not possible to change field names with a label 😥

**Reply by br41nslug** [reply]
Check the message above, this is a closed feature request as it didnt get enough interest from the community. Feel free to create a feature request with the changes you want to see in labels instead of reacting to old, closed and possibly unrelated feature requests.
